### DISCUSSION TITLE
Allow for "Arbitrary attributes" (ex: class="[aria-busy=true]:cursor-progress")

### DISCUSSION THREAD

**Discussion by VicGUTT** [tailwindlabs/tailwindcss#7531] [Ideas]
Hey has anyone ever needed to apply styles based on an element's attribute ?
Wouldn't it be cool if we could take advantage _(abuse ? 🤔)_ of Tailwind's JIT to achieve this ?

Example and use-case:

```html
<button
    onclick="(() => { this.textContent = 'loading...'; this.disabled = true; this.setAttribute('aria-busy', true); })()"
    class="
        bg-indigo-600 text-white text-sm leading-6 font-medium py-2 px-3 rounded-lg shadow-lg
        disabled:opacity-50
        [aria-busy=true]:bg-black
        [aria-busy=true]:cursor-progress
    "
>
    🎉 Click me !
</button>
````

→ [Tailwind Play](https://play.tailwindcss.com/g1EXtJf4IT?file=config)

Here the goal would be to add the `bg-black` and `cursor-progress` styles to the button when it gets the `aria-busy="true"` attribute .

I guess this could also be considerered a "dynamic variant" or an "arbitrary variant".

Food for thought :v:

**Comment by MichaelAllenWarner** [comment]
See also:

- https://github.com/tailwindlabs/tailwindcss/discussions/5659
- https://github.com/tailwindlabs/tailwindcss/discussions/5473
- https://github.com/tailwindlabs/tailwindcss/discussions/6630
